test/e2e.test.ts

```diff
@@ -0,0 +1,279 @@
+  // - 1 GNO = 400 USDC
+  const usdPerCow = ethers.utils.parseEther("0.15");
+  const gnoPrice = usdPerCow.div(400);
+  // - 1 WETH = 4000 USDC
+  const nativeTokenPrice = usdPerCow.div(4000);
+  const priceDenominator = ethers.utils.parseEther("1");
+  const wethBalanceOfUser = ethers.utils.parseUnits("234", 18);
+  let deploymentData: DeploymentData;
+  let vCowTokenSupply: BigNumber;
+  let provenClaims: ProvenClaims;
+  let claim: Claim;
+  let claims: ProvenClaim[];
+
+  it("User Option: claims the user option and vest it", async () => {
+    claim = {
+      account: user.address,
+      claimableAmount: ethers.utils.parseUnits("1234", 18),
+      type: ClaimType.UserOption,
+    };
+    provenClaims = computeProofs([claim]);
+    claims = provenClaims.claims;
+
+    const deploymentParameters: DeploymentParameters = {
+      deployer,
+      cowDao,
+      communityFundsTarget,
+      investorFundsTarget,
+      teamController,
+      provenClaims,
+      usdcPrice,
+      gnoPrice,
+      nativeTokenPrice,
+      usdPerCow,
+      initialCowSupply,
+    };
+    deploymentData = await standardDeployment(deploymentParameters);
+
+    const wethToPay = claim.claimableAmount
+      .mul(nativeTokenPrice)
+      .div(priceDenominator);
+    await deploymentData.wethToken
+      .connect(deployer)
+      .mint(user.address, wethBalanceOfUser);
+    // Claim vCowTokens
+    await deploymentData.wethToken
+      .connect(user)
+      .approve(deploymentData.vCowToken.address, wethToPay);
+    await deploymentData.vCowToken
+      .connect(user)
+      .claim(...getClaimInput(fullyExecuteClaim(claims[0])));
+    vCowTokenSupply = await deploymentData.vCowToken.totalSupply();
+    expect(vCowTokenSupply).to.be.equal(claim.claimableAmount);
+
+    expect(await deploymentData.wethToken.balanceOf(user.address)).to.be.equal(
+      wethBalanceOfUser.sub(wethToPay),
+    );
+    expect(await deploymentData.vCowToken.balanceOf(user.address)).to.be.equal(
+      claim.claimableAmount,
+    );
+    await expect(
+      deploymentData.vCowToken.connect(user).swapAll(),
+    ).to.be.revertedWith("ERC20: transfer amount exceeds balance");
+
+    // Send cowTokens to the vCowToken
+    await deploymentData.cowToken
+      .connect(cowDao)
+      .transfer(deploymentData.vCowToken.address, claim.claimableAmount.div(2));
+
+    // Perform a swapAll
+    const vestingPeriod =
+      await deploymentData.vCowToken.VESTING_PERIOD_IN_SECONDS();
+    setTime(deploymentData.deploymentTimestamp + vestingPeriod / 2);
+    await deploymentData.vCowToken.connect(user).swapAll();
+    expect(await deploymentData.cowToken.balanceOf(user.address)).to.be.equal(
+      claim.claimableAmount.div(2),
+    );
+    expect(
+      await deploymentData.cowToken.balanceOf(deploymentData.vCowToken.address),
+    ).to.be.equal(0);
+    expect(await deploymentData.vCowToken.totalSupply()).to.be.equal(
+      vCowTokenSupply.sub(claim.claimableAmount.div(2)),
+    );
+  });
```

Comment:
This is a test I'd expect and is missing:
```suggestion
    expect(
      await deploymentData.vCowToken.balanceOf(user.address),
    ).to.equal(claim.claimableAmount.div(2));
```
(Maybe then it makes sense to only wait one fourth of the vesting period, so we can test with different numbers that this is equal to 3/4 of the total and the amount taken by the user is 1/4.)